tests/baselines/reference/typeSatisfaction_errorLocations1.errors.txt

```diff
@@ -0,0 +1,146 @@
+    Type 'number' is not assignable to type 'true'.
+
+
+==== typeSatisfaction_errorLocations1.ts (22 errors) ====
+    const obj1 = { a: 1 };
+    
+    const fn1 = (s: { a: true }) => {};
+    fn1({} satisfies unknown);
+        ~~
+!!! error TS2345: Argument of type '{}' is not assignable to parameter of type '{ a: true; }'.
+!!! error TS2345:   Property 'a' is missing in type '{}' but required in type '{ a: true; }'.
+!!! related TS2728 typeSatisfaction_errorLocations1.ts:3:19: 'a' is declared here.
+    fn1({ a: 1 } satisfies unknown);
+          ~
+!!! error TS2322: Type 'number' is not assignable to type 'true'.
+!!! related TS6500 typeSatisfaction_errorLocations1.ts:3:19: The expected type comes from property 'a' which is declared here on type '{ a: true; }'
+    fn1(obj1 satisfies unknown);
+        ~~~~
+!!! error TS2345: Argument of type '{ a: number; }' is not assignable to parameter of type '{ a: true; }'.
+!!! error TS2345:   Types of property 'a' are incompatible.
+!!! error TS2345:     Type 'number' is not assignable to type 'true'.
+    
+    class Cls1 {
+      constructor(p: { a: true }) {}
+    }
+    new Cls1({} satisfies unknown);
+             ~~
+!!! error TS2345: Argument of type '{}' is not assignable to parameter of type '{ a: true; }'.
+!!! error TS2345:   Property 'a' is missing in type '{}' but required in type '{ a: true; }'.
+!!! related TS2728 typeSatisfaction_errorLocations1.ts:9:20: 'a' is declared here.
+    new Cls1({ a: 1 } satisfies unknown);
+               ~
+!!! error TS2322: Type 'number' is not assignable to type 'true'.
+!!! related TS6500 typeSatisfaction_errorLocations1.ts:9:20: The expected type comes from property 'a' which is declared here on type '{ a: true; }'
+    new Cls1(obj1 satisfies unknown);
+             ~~~~
+!!! error TS2345: Argument of type '{ a: number; }' is not assignable to parameter of type '{ a: true; }'.
+!!! error TS2345:   Types of property 'a' are incompatible.
+!!! error TS2345:     Type 'number' is not assignable to type 'true'.
+    
+    function fn2<T extends { a: true }[]>(f: (...args: T) => void) {
+      f({ a: true } satisfies unknown);
+        ~~~~~~~~~~~
+!!! error TS2345: Argument of type '[{ a: boolean; }]' is not assignable to parameter of type 'T'.
+!!! error TS2345:   'T' could be instantiated with an arbitrary type which could be unrelated to '[{ a: boolean; }]'.
+      const o = { a: true as const };
+      f(o satisfies unknown);
+        ~
+!!! error TS2345: Argument of type '[{ a: true; }]' is not assignable to parameter of type 'T'.
+!!! error TS2345:   '[{ a: true; }]' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint '{ a: true; }[]'.
+    }
+    
+    const tuple1: [boolean, boolean] = [true, 100 satisfies unknown];
+                                                  ~~~~~~~~~
+!!! error TS2322: Type 'number' is not assignable to type 'boolean'.

```

Comment:
This behavior is unchanged by this PR, but looks wrong. Did you look at this example?

Reply:
heh, I was starring at this and somehow missed the fact that it's wrong in the very same way 😅 

I just pushed out a fix for this. It is a little bit cumbersome to call `getEffectiveCheckNode` around the code to get the best behavior... I think it semantically makes a lot of sense. It is the expression of the `SatisfiesExpression` that we want to highlight. But since I have to call it at multiple places it's rather easy to miss some places where it should be called too.

So I wonder now if it wouldn't be better to flip the situation slightly. I could move this logic into `getErrorSpanForNode` and "target" the `satisfies` token in `checkSatisfiesExpressionWorker` (since that's the only place where we want to target the `satisfies` token). The problem is... that I can't easily pass a token down as `errorNode`. It's not even directly accessible in the checker. So maybe a text span could be passed down? I'm not sure if this is even feasible though since that argument is passed down through some functions and they might even try to check stuff on it. So overall it's likely not a 5min change.

Do you think the current solution with `getEffectiveCheckNode` is OK? Or should I try this other alternative?

Reply:
I think this is fine.